File: No_1695/Maximum Erasure Value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int ret = 0;
        int sum = 0;
        int left = 0, right = 0, n = nums.size();
        unordered_set<int> hset;
        
        while(right < n) {
            if(!hset.count(nums[right])) {
                hset.insert(nums[right]);
                sum += nums[right];
                ret = max(ret, sum);
                right++;
            } else {
                sum -= nums[left];
                hset.erase(nums[left]);
                left++;
            }
        }
        
        return ret;
    }
};

File: No_1695/Maximum Erasure Value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        seen = set()
        left = 0
        cur_sum = 0
        ans = 0

        for right in range(len(nums)):
            while nums[right] in seen:
                seen.remove(nums[left])
                cur_sum -= nums[left]
                left += 1

            seen.add(nums[right])
            cur_sum += nums[right]
            ans = max(ans, cur_sum)

        return ans